File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    maxCandy = Math.max(...candies)
    candyBooleans = candies.map((candy) => {
        return candy + extraCandies >= maxCandy
    })
    return candyBooleans
};

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.rb
# @param {Integer[]} candies
# @param {Integer} extra_candies
# @return {Boolean[]}
def kids_with_candies(candies, extra_candies)
    max_candy = candies.max

    new_array = candies.map do |candy|
        candy + extra_candies >= max_candy
    end

    new_array

end